src/common/os.cc

```diff
@@ -147,6 +147,14 @@ namespace Polling {
        epoll_fd = TRY_RET(epoll_create(Const::MaxEvents));
     }
 
+    Epoll::~Epoll()
+    {
+        if (epoll_fd > 0)
```

Comment:
While it is unlikely that epoll_fd would ever be '0', this is technically possible.  The caller could close STDIN, then invoke a code path that creates an Epoll instance.  You should test for ">= 0".

Reply:
Yes, he's right. File descriptors are suppose to be always allocated from the lowest available ordinal.

Reply:
Yap you are right, changes applied